diff --git a/src/flask/cli.py b/src/flask/cli.py
index 5b3116e..1a9159e 100644
--- a/src/flask/cli.py
+++ b/src/flask/cli.py
@@ -777,7 +777,7 @@ def show_server_banner(debug: bool, app_import_path: str | None) -> None:
         click.echo(f" * Debug mode: {'on' if debug else 'off'}")
 
 
-class CertParamType(click.ParamType[str | os.PathLike[str] | ssl.SSLContext]):
+class CertParamType(click.ParamType[t.Any]):
     """Click option type for the ``--cert`` option. Allows either an
     existing file, the string ``'adhoc'``, or an import for a
     :class:`~ssl.SSLContext` object.
